File: backend/GameApp.Domain/ValueObjects/Items/ItemDescription.cs
namespace GameApp.Domain.ValueObjects.Items;


// Value Object for a game's inventory item description
public class ItemDescription
{
    private static readonly string _messageIfEmpty = "Item description cannot be empty";
    
    private readonly string Description;

    public ItemDescription(string value)
    {
        if (string.IsNullOrWhiteSpace(value))
            throw new ArgumentException(_messageIfEmpty, nameof(value));

        Description = value;
    }

    public string GetDescription() => Description;

    public ItemDescription SetDescription(string newDescription)
    {
        if (string.IsNullOrWhiteSpace(newDescription))
            throw new ArgumentException(_messageIfEmpty, nameof(newDescription));

        return new ItemDescription(newDescription);
    }

    public override bool Equals(object? obj)
    {
        if (obj is not ItemDescription other)
            return false;

        return Description == other.Description;
    }

    public override int GetHashCode() => Description.GetHashCode();

    public override string ToString() => Description;
}

File: frontend/gameapp-frontend/src/domain/value-objects/items/item-description.ts
// Value Object for item description
export class ItemDescription {
    static ERROR_EMPTY = "Item description cannot be empty";

    private _description: string;

    constructor(description: string) {
        this._description = description;
    }

    get description(): string {
        return this._description;
    }

    setDescription(newDescription: string): ItemDescription {
        return new ItemDescription(newDescription);
    }

    equals(other: unknown): boolean {
        return other instanceof ItemDescription &&
               this._description === other._description;
    }

    toString(): string {
        return this._description;
    }
}
